src/react-components/home/HomePage.js

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useEffect } from "react";
+import { FormattedMessage, addLocaleData } from "react-intl";
+import en from "react-intl/locale-data/en";
+import classNames from "classnames";
+import configs from "../../utils/configs";
+import IfFeature from "../if-feature";
+import { Page } from "../layout/Page";
+import MediaTiles from "../media-tiles";
+import { CreateRoomButton } from "../input/CreateRoomButton";
+import { PWAButton } from "../input/PWAButton";
+import { useFeaturedRooms } from "./useFeaturedRooms";
+import styles from "./HomePage.scss";
+import mediaBrowserStyles from "../../assets/stylesheets/media-browser.scss";
+import discordLogoSmall from "../../assets/images/discord-logo-small.png";
+import { AuthContext } from "../auth/AuthContext";
+import { createAndRedirectToNewHub } from "../../utils/phoenix-utils";
+
+addLocaleData([...en]);
+
+export function HomePage() {
+  const featuredRooms = useFeaturedRooms();
+  const auth = useContext(AuthContext);
+
+  useEffect(() => {
+    const qs = new URLSearchParams(location.search);
+
+    // Support legacy sign in urls.
+    if (qs.has("sign_in")) {
```

Comment:
What actually uses this sign in url?

Reply:
I don't actually know, I just copy pasted that.